File: lib/models/stained_glass.dart
class StainedGlass {
  final String title;
  final String locationId;
  final List<String> informationIds;

  StainedGlass({
    required this.title,
    required this.locationId,
    required this.informationIds,
  });

  Map<String, dynamic> toJson() {
    return {
      'title': title,
      'locationId': locationId,
      'informationIds': informationIds,
    };
  }

  factory StainedGlass.fromMap(Map<String, dynamic> map) {
    return StainedGlass(
      title: map['title'],
      locationId: map['location'],
      informationIds: List<String>.from(map['info'] ?? []),
    );
  }
}

File: android/app/src/main/java/com/google/ar/core/examples/java/augmentedimage/StainedGlass.java
package com.example.vitral_app;
import java.util.List;
import java.util.Map;
import java.util.HashMap;
import java.util.ArrayList;


public class StainedGlass {
    private String title;
    private String locationId;
    private List<String> informationIds;

    public StainedGlass(String title, String locationId, List<String> informationIds) {
        this.title = title;
        this.locationId = locationId;
        this.informationIds = informationIds;
    }

    public String getTitle() {
        return title;
    }

    public String getLocationId() {
        return locationId;
    }

    public List<String> getInformationIds() {
        return informationIds;
    }

    // Convert StainedGlass to JSON (map)
    public Map<String, Object> toJson() {
        Map<String, Object> map = new HashMap<>();
        map.put("title", title);
        map.put("locationId", locationId);
        map.put("informationIds", informationIds);
        return map;
    }

    // Create a StainedGlass instance from a map
    public static StainedGlass fromMap(Map<String, Object> map) {
        String title = (String) map.get("title");
        String locationId = (String) map.get("location");
        List<String> informationIds = (List<String>) map.getOrDefault("info", new ArrayList<>());
        return new StainedGlass(title, locationId, informationIds);
    }
}
